lib/Games/Dice/Advanced.pm

```diff
@@ -148,7 +154,8 @@ sub new {
 	            $mul * _sum(map { 1 + int(rand($faces)) } (1..$repeats))
 	        };
 	    } elsif(ref($recipe) eq 'CODE') {
-	        $self = sub { $mul * &{$recipe} }
+	        my $r = &{$recipe};
+	        $self = $r =~ /^\d+$/ ? sub { $mul * $r } : sub { $r };
```

Comment:
I wonder if it might be reasonable to just not bother checking and if the user tries to multiply B♭ by 7 they deserve all the pain they get. Alternatively, make doing that _fatal_ instead of just not multiplying. In code this old it was common to try hard to grok what the user wanted, but more modern me thinks it's better to fail fast and hard (and that all code should be written in Idris or Rust 🫤).

If this check is kept then it needs a test - in which case the check for a number should be something like `$r =~ /^-?\d+(\.\d+)?(e\d+)$/i` to cope with negatives and floats.

Reply:
Yep. My RE is very naive.